message/msg_addr.hh

```diff
@@ -10,18 +10,22 @@
 
 #include "gms/inet_address.hh"
 #include <cstdint>
+#include "locator/host_id.hh"
 
 namespace netw {
 
 struct msg_addr {
     gms::inet_address addr;
     uint32_t cpu_id;
-    friend bool operator==(const msg_addr& x, const msg_addr& y) noexcept;
-    friend bool operator<(const msg_addr& x, const msg_addr& y) noexcept;
-    friend std::ostream& operator<<(std::ostream& os, const msg_addr& x);
+    locator::host_id host_id;
+    msg_addr(gms::inet_address ip, uint32_t cpu = 0, std::optional<locator::host_id> opt_id = std::nullopt) noexcept
```

Comment:
Why not ditch `std::optional` and default to `null_id` instead?

Reply:
That's possible.

IIRC initially I planned to pass an optional that might be disengaged from `messaging_service::get_source`
but `retrieve_auxiliary_opt` returns a pointer, not optional so this turned out to be a moot point.